[waylonggggg] iP

waylonggggg opened the pull request:
# Sonder
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Sonder frees your mind of having to remember the things you got to do! It's
- text-based
- easy to learn
- ~~fast~~ SUPER FAST to use

All you have to do is,
1. Download it from [here](https://github.com/waylonggggg/ip)
2. Run it
3. Add your tasks with our easy to use CLI :fire::fire::fire:

And it is completely **FREE!**

Features:
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (Coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

waylonggggg committed:
Greeting Message

waylonggggg committed:
Add echo function

waylonggggg committed:
Add list functionality

waylonggggg committed:
Add ability to mark and unmark tasks as done

waylonggggg committed:
Sonder.java: Refactor long if-else block to switch block with helper function

waylonggggg committed:
Add todo, deadline and event class

waylonggggg committed:
Modify input.txt for automated testing

waylonggggg committed:
Add custom exception class for invalid inputs

waylonggggg committed:
Add delete task function

waylonggggg committed:
Add save locally feature

waylonggggg committed:
Merge branch 'branch-Level-7'

waylonggggg committed:
Add date for tasks

waylonggggg committed:
Merge branch 'branch-Level-8'

waylonggggg committed:
Implement more OOP

waylonggggg committed:
Add more OOP

waylonggggg committed:
Merge branch 'branch-A-MoreOOP'

waylonggggg committed:
Add all classes into one package

waylonggggg committed:
Merge branch 'add-gradle-support'

waylonggggg committed:
Add gradle

waylonggggg committed:
Modify chatbot intro message

waylonggggg committed:
Merge branch 'branch-Level-8'

waylonggggg committed:
Add JUnit tests

waylonggggg committed:
Add JavaDoc

waylonggggg committed:
Merge branch 'branch-A-JavaDoc'

waylonggggg committed:
Improve code quality

waylonggggg committed:
Merge branch-A-CodingStandard

CVincent0907 commented on src/main/java/sonder/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class Deadline extends Task {
+
+    private LocalDate by;
```
Suggestion: The choice of name may not be so understandable by others. Perhaps, you could try `private LocalDate dueDate`.

CVincent0907 commented on src/main/java/sonder/Parser.java:
```diff
@@ -0,0 +1,196 @@
+
+public class Parser {
+
+    private TaskList tasks;
```
You might want to choose a singular noun for `private Tasklist` (ie: taskList) since it is not a collection of tasks now as you have just implemented a `Tasklist` class to keep track of all the tasks.

CVincent0907 reviewed (commented):
LSTM

waylonggggg committed:
Removed one line of code

waylonggggg committed:
Add find functionality

waylonggggg committed:
Merged branch-Level-9

waylonggggg committed:
Update README.md

pranavyey commented on src/main/java/sonder/Parser.java:
```diff
@@ -0,0 +1,200 @@
+
+    public void run(String input, String[] inputArr, String command, int length) throws SonderException,
+            FileNotFoundException, IOException {
+        switch (command) {
```
Good use of switch and case!

pranavyey reviewed (commented):
Overall, good work! Coding standards upheld

pranavyey commented on src/main/java/sonder/Parser.java:
```diff
@@ -0,0 +1,200 @@
+        }
+
+        String[] split = input.split("/by");
+        String taskDescription = split[0].substring(9).trim();
```
Why is there a need to limit to the first 9 or 10 characters of the string?

pranavyey commented on src/main/java/sonder/Parser.java:
```diff
@@ -0,0 +1,200 @@
+    }
+
+    private void addTodoTask(String input) throws IOException {
+        String taskDescription = input.substring(5).trim();
```
Why is there a need to limit to first 5 characters of the string?

pranavyey commented on src/main/java/sonder/Sonder.java:
```diff
@@ -0,0 +1,66 @@
+        this.parser = new Parser(tasks, ui, storage);
+    }
+
+    private TaskList initialiseTaskList() {
```
can this method be abstracted out to the TaskList class?

pranavyey commented on src/main/java/sonder/Sonder.java:
```diff
@@ -0,0 +1,66 @@
+        }
+    }
+
+    private void handleException(Exception e) {
```
consider using switch and case here because of the number of possible scenarios, which could increase in the future

pranavyey commented on src/main/java/sonder/Storage.java:
```diff
@@ -0,0 +1,199 @@
+package sonder;
+
+import java.io.*;
```
is it necessary to import java.io.*? can we import less packages instead?

pranavyey reviewed (commented):
Overall good work!

waylonggggg committed:
Add checkstyle

waylonggggg committed:
Merge branch 'branch-A-CheckStyle'

waylonggggg committed:
Merge branch 'master' of https://github.com/waylonggggg/ip

waylonggggg committed:
Add Gui

waylonggggg committed:
Merge branch 'branch-Level-10'

waylonggggg committed:
Correct Readme

waylonggggg committed:
Fix delete bug

waylonggggg committed:
Add assertions to all files

All class files do not have assertions for testing.

Assertions are great for developer testing.

Let's add assertion to the methods in classes that are applicable.

Assertions help to ensure that wrongful parameters are not passed into methods.

waylonggggg committed:
Merge pull request #1 from waylonggggg/branch-A-Assertions

Add assertions to all files

waylonggggg committed:
Improve coding standard

There are several parts of the code that can be refactored.

Refactoring the code improves it's readability.

Let's do some extracting and consolidating.

This helps to reduce redundant code.

waylonggggg committed:
Merge branch 'master' into branch-A-CodeQuality

waylonggggg committed:
Merge pull request #2 from waylonggggg/branch-A-CodeQuality

Improve coding standard

waylonggggg committed:
Add a better search functionality

In order to find a task it's full name had to be typed out.

This is inefficient for users especially if the task name is long.

Let's implement a find command that can list out tasks even if the keyword matches the task partially.

This leads to a more efficient search system.

waylonggggg committed:
Merge pull request #3 from waylonggggg/branch-C-BetterSearch

Add a better search functionality

waylonggggg committed:
Add user guide

waylonggggg committed:
Update README.md
